File: r_csv/assignment2.R
library(rvest)
library(readr)

# Define the URL/ website
url <- "https://en.wikipedia.org/wiki/Comma-separated_values"

# Read the webpage/website
tables <- read_html(url)

# Extract all tables
extract_tables <-html_nodes(tables, "table")

# Extract the car table
cars_tables <- html_table(extract_tables[[2]], fill = TRUE)
print(cars_tables)

# Save the cars table as CSV file
write.csv(cars_tables, file = "~/Desktop/st_2195/st2195_assignment_2/r_csv/cars.csv", row.names = FALSE)

File: python_csv/Assignment2.py
import requests
import pandas as pd
from bs4 import BeautifulSoup
import io

#Define the URL
url = "https://en.wikipedia.org/wiki/Comma-separated_value"
# Get the url content
response = requests.get(url)
soup = BeautifulSoup(response.text, "html.parser")

# Extract all tables from the url
tables = pd.read_html(io.StringIO(str(soup)))

# Extract cars table
cars = tables[1]
print(cars)

# Export cars to csv
cars.to_csv("/Users/rukhshandaashraf/Desktop/st_2195/st2195_assignment_2/python_csv/cars_python.csv",index=False)
